src/quotes/generate.js

```diff
@@ -0,0 +1,88 @@
+import partialRight from 'lodash.partialright';
+import unzip from 'lodash.unzip';
+import isEmpty from 'lodash.isempty';
+import isEqual from 'lodash.isequal';
+import { ROLES, PREFIX, PAST, FUTURE, ADVERBS, ADJECTIVES, NOUNS } from './quotes';
+
+export function getQuoteFromID(role, action, quoteID) {
+  const digits = convertToDigits(quoteID);
+  if (isEmpty(digits)) {
+    return;
+  }
+  const roleLeaf = ROLES[role];
+  const prefixLeaf = ROLES[PREFIX][action];
+
+  // this order is not fluid and affects proper encoding, decoding, refactor by encoding this list
+  // const digits = [prefixPast, rolePast, prefixFuture, roleFuture];
+  const arrayExpressions = [prefixLeaf[PAST], roleLeaf[PAST], prefixLeaf[FUTURE], roleLeaf[FUTURE]];
+
+  const expressions = arrayExpressions
+    // module is if we somehow run out of bounds, it should not happen, but it will map to something
+    .map((phrases, index) => phrases[digits[index] % phrases.length])
+    .join(' ');
+  return expressions;
+}
+
+export function getRandomQuoteID(role, action) {
+  const [, quoteID] = getRandomQuoteAndID(role, action);
+  return convertToNickname(quoteID);
+}
+
+function getRandomQuoteAndID(role, action) {
+  const roleLeaf = ROLES[role];
+  const prefixLeaf = ROLES[PREFIX][action];
+
+  // this order is not fluid and affects proper encoding, decoding, refactor by encoding this list
+  const arrayOfExpressions = [prefixLeaf[PAST], roleLeaf[PAST], prefixLeaf[FUTURE], roleLeaf[FUTURE]];
+
+  const sampleAndIndex = items => {
+    // index as string, to cover for 0
+    const index = Math.floor(Math.random() * items.length);
+    const item = items[index];
+    return [item, index];
+  };
+
+  const [expressions, index] = unzip(arrayOfExpressions.map(phrases => sampleAndIndex(phrases)));
+  const expression = expressions.join(' ');
+  return [expression, index];
+}
+
+// URL-NICKNAME GENERATING
```

Comment:
Way too many comments for me. Not that I'm against comments, but for consistency I would rather remove most of them since there are only few in the code-base and those that are left serve different purpose. I'm open for a discussion 😎

Reply:
sure, this one in particular I would keep, but will try to put my internal novelist to sleep :P